animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -241,21 +273,38 @@ cdef class Mesh:
             return self.polygons.indices.length
 
     def appendAttributes(self, Mesh source):
```

Comment:
You can probably add a method that appends a single attribute, then in this method, loop explicitly over each of the dictionaries and call the newly created method.